contracts/upgradable/Proxy.sol

```diff
@@ -13,7 +13,6 @@ contract Proxy is BaseProxy {
         bytes memory setupParams
     ) {
         if (owner == address(0)) revert InvalidOwner();
-        if (implementation() != address(0)) revert AlreadyInitialized();
```

Comment:
Why is this check removed?

Reply:
It seems like this check would never fail because it is called in the constructor and implementation() would always return the zero address. Let me know if you can think of a case where it would fail.